### DISCUSSION TITLE
Event when player leftclicks with item

### DISCUSSION THREAD

**Discussion by DasLixou** [Minestom/Minestom#988] [Help]
hi! i need an event which gets called when a player leftclicks and i wanna also get the used item.

I'm currently using `PlayerHandAnimationEvent` but that get's also called when i place a block. 

Any alternative or filtering between block is placed or leftclicked?

**Comment by ArvedEh** [comment]
A player left-clicking a block is actually a `PlayerStartDiggingEvent`.
To check if a player right-clicked (anything), you can also use the `PlayerUseItemEvent` (but I don't know if the client sends that packet when you right-click without an item in your hand)

**Reply by DasLixou** [reply]
but PlayerStartDigging doenst get called when left-clicking in the air xD I just need an alternative to spigots InteractEvent with LEFTCLICK_AIR and LEFTCLICK_BLOCK xD

**Reply by Eoghanmc22** [reply]
lookes like spigot just checks if the player is looking at air when it gets an animation event. see https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/nms-patches/net/minecraft/server/network/PlayerConnection.patch#1099

**Reply by DasLixou** [reply]
Yes but in Minestom there is no way to get the action I guess. Maybe I’ll make a pr about it
